File: HTML/resume/resumeJSON.js
// for skills just to have an example of how it works
var xhr = new XMLHttpRequest();
xhr.open("GET", "resumeJSON.json", true);
xhr.onreadystatechange = function () {
  if (xhr.readyState === 4 && xhr.status === 200) {
    var data = JSON.parse(xhr.responseText);
    var ss = data.skills;
    var os = Object.values(ss);
    var ulSkills = document.querySelector("section.skills>div.section-body>ul");
    os.forEach((s) => {
      let skillItem = document.createElement("li");
      skillItem.innerHTML = s.title + ": " + s.desc;
      ulSkills.appendChild(skillItem);
    });
  }
};
xhr.send();

// starting from the top. this is header:

var xhr = new XMLHttpRequest();
xhr.open("GET", "JSONresume.json", true);
xhr.onreadystatechange = function () {
  if (xhr.readyState === 4 && xhr.status === 200) {
    var data = JSON.parse(xhr.responseText);

    var dobItem = document.createElement("li");
    dobItem.innerHTML =
      '<i class="fa-solid fa-calendar"></i>: ' + data.header.DoB; // DoB is case sensitive
    document.querySelector("ul.contact-info").appendChild(dobItem);
    var phoneItem = document.createElement("li");
    phoneItem.innerHTML =
      '<i class="fa-solid fa-phone"></i>: ' + data.header.phone;
    document.querySelector("ul.contact-info").appendChild(phoneItem);

    var mailItem = document.createElement("li");
    mailItem.innerHTML = '<i class="fa-solid fa-at"></i>: ' + data.header.email;
    document.querySelector("ul.contact-info").appendChild(mailItem);

    var gitItem = document.createElement("li");
    gitItem.innerHTML =
      '<i class="fa-brands fa-github"></i>: ' + data.header.git;
    document.querySelector("ul.contact-info").appendChild(gitItem);

    var liItem = document.createElement("li");
    liItem.innerHTML =
      '<i class="fa-brands fa-linkedin"></i>: ' + data.header.linkedin;
    document.querySelector("ul.contact-info").appendChild(liItem);

    var flItem = document.createElement("li");
    flItem.innerHTML =
      '<i class="fa-brands fa-python"></i>: ' + data.header.freelance;
    document.querySelector("ul.contact-info").appendChild(flItem);

    var name = document.querySelector("div.name");
    name.innerHTML = data.name;

    var coverLetterText = document.querySelector(".cover-letter>.section-body");
    coverLetterText.innerHTML = data.coverLetter;

    var aboutText = document.querySelector(".about>.section-body");
    aboutText.innerHTML = data.about;

    var projectsList = document.querySelector("ul.projects-ul");
    projectsData = data.projects;
    projectsDataObject = Object.values(projectsData);
    projectsDataObject.forEach((o) => {
      listItem = document.createElement("li");
      title = o.title;
      desc = o.desc;
      titleItem = document.createElement("h3");
      titleItem.innerHTML = title;
      listItem.appendChild(titleItem);
      descItem = document.createElement("p");
      descItem.innerHTML = desc;
      listItem.appendChild(descItem);
      projectsList.appendChild(listItem);
    });

    var skillsList = document.querySelector("ul.skills-ul");
    skillsData = data.skills;
    skillsDataObject = Object.values(skillsData);
    skillsDataObject.forEach((o) => {
      listItem = document.createElement("li");
      title = o.title;
      desc = o.desc;
      titleItem = document.createElement("h3");
      titleItem.innerHTML = title;
      listItem.appendChild(titleItem);
      descItem = document.createElement("p");
      descItem.innerHTML = desc;
      listItem.appendChild(descItem);
      skillsList.appendChild(listItem);
    });

    var expList = document.querySelector(".exp-ul");
    expData = data.experience;
    expDataO = Object.values(expData);
    expDataO.forEach((o) => {
      listItem = document.createElement("li");
      title = o.title;
      company = o.company;
      time = o.time;
      titleItem = document.createElement("h4");
      titleItem.innerHTML = title;
      listItem.appendChild(titleItem);
      companyItem = document.createElement("h5");
      companyItem.innerHTML = company;
      listItem.appendChild(companyItem);
      timeItem = document.createElement("p");
      timeItem.innerHTML = time;
      listItem.appendChild(timeItem);
      expList.appendChild(listItem);
    });

    var eduUL = document.querySelector("ul.edu-ul");
    eduData = data.education;
    eduDataObject = Object.values(eduData);
    eduDataObject.forEach((o) => {
      listItem = document.createElement("li");
      h4Item = document.createElement("h4");
      h4Item.innerHTML = o.title + " - " + o.institution;
      listItem.appendChild(h4Item);
      listItem.append(o.desc);
      eduUL.appendChild(listItem);
    });
    var hobbiesUL = document.querySelector(".hobbies-ul");
    hobbiesData = data.hobbies;
    hobbiesDataObject = Object.values(hobbiesData);
    hobbiesDataObject.forEach((o) => {
      listItem = document.createElement("li");
      listItem.innerHTML = "<h3>" + o.title + "</h3> " + o.desc;
      hobbiesUL.appendChild(listItem);
    });

    var certsDiv = document.querySelector("div.certs-body");
    certs = data.certificates;
    certsO = Object.values(certs);
    certsO.forEach((o) => {
      imge = document.createElement("img");
      imge.src = o.src;
      imge.alt = o.alt;
      imge.style = "width:25rem;";
      certsDiv.appendChild(imge);
    });
  }
};
xhr.send();

File: HTML/resume/resumeJSON.py
import json

# this is the file to control the data inside the json file. 
# do not touch the file JSONresume.json manually, only modify it HERE!



data = {}
data = {
    "name":"Itamar Barkai",
  "header": {
    "name": "Itamar Barkai",
    "age": 24,
    "DoB": "1998-02-19",
    "email": "ibark.bu@gmail.com",
    "phone": "+972 - 52 - 641 - 2626",
    "git": "https://github.com/TheAllSeer",
    "linkedin": "https://www.linkedin.com/in/itamar-barkai-26119a226/",
    "freelance": "https://www.upwork.com/freelancers/~01b87343c472baed1e",
    "cover": "I know that the world is going to continue evolving into a more technology oriented area, where inevitably people will feel more lonely. social media is a powerful tool to combat this phenomenon, and I want to help make the situation better."
  }
}
data['about'] = "I am a passionate full-stack developer who specializes in Python. I am a quick learned looking to challenge myself by working on interesting and important projects. my emphasis is on the project's end goal more than the programming language I use."

data['projects'] = {
    0:{
      "title": "Facebook Marketplace Scraper",
      "desc": "the scraper searches the marketplace for listings that match specific criteria such as price range, category, and keywords. Once it finds new listings that match these criteria, it sends an email notification to the user. This tool can be used to keep track of new listings that match specific criteria, such as products that fall within a certain price range or keywords that are relevant to your interests. This can be helpful for businesses or individuals who want to stay informed about new products or deals on the marketplace, or for tracking prices and inventory of certain products."
    }
}

data["skills"]={
    0: {
      "title": "Programming Languages",
      "desc": "Python, SQL, JavaScript, HTML, CSS, JSON"
    },
    1: {
      "title": "Tools",
      "desc": "Jupyter-lab, advanced Excel, Visual Studio Code, SSMS, Power BI"
    },
    2: {
      "title": "languages",
      "desc": "native speaker in both Hebrew and English"
    }
  }

data["experience"] = {
    0: {
      "title": "Language Data Analyst",
      "company": "Transperfect",
      "time": "from 2022-08-02 until 2022-12-01"
    },
    1: {
      "title": "Full Stack Developer",
      "company": "Freelance",
      "time": "from 2022-12-02 and going"
    }
  }
data["education"]= {
    0: {
      "title": "High School Diploma",
      "institution": "Hebrew Reali School, Beit Biram",
      "desc": "full high school diploma"
    },
    1: {
      "title": "Pre Academic Program",
      "institution": "Technion - Israeli Institute of Technology",
      "desc": "pre academic program with extensive physics and mathematics learning"
    },
    2: {
      "title": "Psychometric exam",
      "institution": "Generalized National Test",
      "desc": "scored 664, with a mathematics oriented score of 682"
    }
  }


data["certificates"]= {
    "cert1": {
      "alt": "Coursera WD4E complete course",
      "src": "Coursera WD4E complete course.jpg"
    },
    "cert2": {
      "alt": "Coursera Computational thinking Penn",
      "src": "Coursera Computational thinking Penn.jpg"
    },
    "cert3": {
      "alt": "John Bryce Data Analyst Expert",
      "src": "John Bryce_page-0001.jpg"
    }
  }


data["hobbies"]= {
    "hobby1": {
      "title": "Gym",
      "desc": "have been working out for 9 years and have participated in 5 national powerlifting competitions"
    },
    "hobby2": {
      "title": "Guitar",
      "desc": "I've been playing guitar for the past 9 years"
    },
    "hobby3": {
      "title": "Gaming",
      "desc": "I have been playing video games for the past 14 years"
    }
  }
data['coverLetter'] = 'I know that the world is going to continue evolving into a more technology oriented area, where inevitably people will feel more lonely. social media is a powerful tool to combat this phenomenon, and I want to help make the situation better.'
print(json.dumps(data, indent=4))
with open('JSONresume.json', 'w') as file:
    json.dump(data, file)
